src/Castle.Facilities.AspNetCore/Resolvers/FrameworkDependencyResolver.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright 2004-2018 Castle Project - http://www.castleproject.org/
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.Facilities.AspNetCore.Resolvers
+{
+	using System;
+	using System.Linq;
+
+	using Castle.Core;
+	using Castle.MicroKernel;
+	using Castle.MicroKernel.Context;
+
+	using Microsoft.Extensions.DependencyInjection;
+
+	public class FrameworkDependencyResolver : ISubDependencyResolver
+	{
+		private readonly IServiceCollection serviceCollection;
+		private static ServiceProvider serviceProvider;
+
+		public FrameworkDependencyResolver(IServiceCollection serviceCollection)
+		{
+			this.serviceCollection = serviceCollection;
+		}
+
+		public bool CanResolve(CreationContext context, ISubDependencyResolver contextHandlerResolver, ComponentModel model, DependencyModel dependency)
+		{
+			return serviceCollection.HasMatchingType(dependency.TargetType);
+		}
+
+		public object Resolve(CreationContext context, ISubDependencyResolver contextHandlerResolver, ComponentModel model, DependencyModel dependency)
+		{
+			serviceProvider = serviceProvider ?? serviceCollection.BuildServiceProvider();
+			return serviceProvider.GetService(dependency.TargetType);
+		}
+	}
+
+	public static class GenericTypeExtensions
```

Comment:
Is it a good idea making `GenericTypeExtensions` and `ServiceCollectionExtensions` public? Are you using them in unit tests too?

Reply:
No, none at all. Purely muscle memory kicking in here.

Reply:
These are super covered by tests in https://github.com/fir3pho3nixx/Windsor/blob/aspnet-core-windsor-final/src/Castle.Facilities.AspNetCore.Tests/AspNetCoreFacilityTestCase.cs

Reply:
And https://github.com/fir3pho3nixx/Windsor/blob/aspnet-core-windsor-final/src/Castle.Facilities.AspNetCore.Tests/Resolvers/ServiceCollectionExtensionsTestCase.cs#L50

Reply:
> Are you using them in unit tests too?
No, I will move this over ...

Reply:
It is integration tested by:
 - https://github.com/fir3pho3nixx/Windsor/blob/aspnet-core-windsor-final/src/Castle.Facilities.AspNetCore.Tests/AspNetCoreFacilityTestCase.cs
 - https://github.com/fir3pho3nixx/Windsor/blob/aspnet-core-windsor-final/src/Castle.Facilities.AspNetCore.Tests/Resolvers/ServiceCollectionExtensionsTestCase.cs

Reply:
> No, I will move this over ...

No problem. I wasn't worried about them being unit tested, more wondering if they were public because the unit tests also used them.